### ISSUE TITLE
ERROR: leaked a completely closed stream

### ISSUE THREAD

**Issue by vyzo** [libp2p/go-mplex#48]
I am seeing this error quite a bit in the mplex relay:
```
ERROR      mplex: leaked a completely closed stream multiplex.go:324
```

Judging by the comments around the source of the log, this is close to an assertion violation and indicative of a bug.

**Comment by vyzo** [comment]
Possibly related to #47

**Comment by Stebalien** [comment]
It's because we reset the stream _then_ remove it from the set of open streams without holding a lock between.

**Comment by vyzo** [comment]
I think we can just ignore this condition then (or simply downgraded to `Debug`).
Per conversation with @whyrusleeping he agrees that we should simply turn it to debug, it spams quite a bit.

**Comment by Stebalien** [comment]
That's still _wrong_. That is, the debug message is incorrect and pretty much useless. We should either drop the debugging code or track the state correctly.

**Comment by vyzo** [comment]
Fair enough,  I will remove the short-circuit and drop the debugging code altogether.

**Comment by vyzo** [comment]
Addressed in #52.
